articles/i2p-docker-services/FreecivServerClient.md

```diff
@@ -0,0 +1,234 @@
+	eyedeekay/i2p-freeciv
+```
+
+Freeciv over i2p: The Client
+----------------------------
+
+### Prerequisites
+
+Similar to the previous tutorials, except in this case it is intended to be run
+on a computer other than the computer running the server. You still need an i2p
+router running on the client computer.
+
+Particularly, you will need the destination(either the base32 or the
+addresshelper address) of an existing i2p-based Freeciv server. Intermittently,
+there will be one available at the following b32:
+4q7zxgr27pwybbqizpv6nhmriidgwj3owqt33jvuijmro57l7jdq.b32.i2p, but it's not going
+to be up all the time(it's the one I used to test this on my laptop). I'm going
+to use it in the example.
+
+### Setting up i2pd
+
+We haven't configured a client tunnel in this series of guides yet. Let's do
+that now.
+
+#### Client tunnel configuration
+
+Clients will need a destination to connect to, corresponding to a server like
+the one we already set up. In this case, the client tunnel should also listen on
+all addresses so other containers on the docker network can use it to connect to
+the Freeciv server.
+
+```
+[FREECIV-CLIENT]
+type = client
+address = 0.0.0.0
+port = 5555
+destination = 4q7zxgr27pwybbqizpv6nhmriidgwj3owqt33jvuijmro57l7jdq.b32.i2p
+inbound.length = 1
+inbound.quantity = 15
+outbound.length = 1
+outbound.quantity = 15
+keys = freeciv-client.dat
+```
+
+You'll notice that most of the options are the same between the server and
+client. The big difference, the destination option, is set to the base32 of a
+freeciv server running on i2p.
+
+### Setting up Freeciv
+
+In this example, we're using the freeciv-client-gtk3 client to connect to the
+Freeciv server. The instructions below can at least be adapted to the SDL and
+QT versions, possibly all versions of the Freeciv client.
+
+#### Freeciv Client Configuration
+
+In order to successfully connect to the Freeciv server through i2p, use the
+newly created client tunnel host:port with the --autoconnect option in your
+final CMD in your Dockerfile.
+
+```
+/usr/games/freeciv-gtk3 \
+    --autoconnect \
+    --server 172.81.81.2 \
+    --port 5555
+```
+
+#### The Dockerfile
+
+Is extremely self-explanatory.
+
+```Dockerfile
+FROM debian:sid
+RUN apt-get update
+RUN apt-get install -y freeciv-client-gtk3
+RUN adduser --disabled-password --gecos ',,,,' freeciv
+USER freeciv
+CMD /usr/games/freeciv-gtk3 \
+    --autoconnect \
+    --server 172.81.81.2 \
+    --port 5555
+```
+
+You can build it with:
+
+```
+docker build --rm \
+	-f Dockerfile.client -t eyedeekay/i2p-freeciv-client .
+```
+
+and run it by forwarding to your local X server with:
+
+```
+docker run -i -t --rm \
+	-e DISPLAY=:0 \
+	--name i2p-freeciv-client \
+	--network eepsite \
+	--network-alias i2p-freeciv-client \
+	--hostname i2p-freeciv-client \
+	--ip 172.81.81.4 \
+	--link freeciv-i2pd \
+	--volume freeciv-client:/home/freeciv \
+	--volume /tmp/.X11-unix:/tmp/.X11-unix:ro \
+	eyedeekay/i2p-freeciv-client
+```
```

Comment:
This one could do with a brief conclusion, such as "And now for the last step, finding some players! Join r/i2p or ask around on IRC!", it would help to round out the guide.

Reply:
Looks like github missed my edit. I put a conclusion under its own heading at the bottom.